### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by odaialazzeh**
- [ ] Create a Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign-up and log-in pages.

- [ ] Create a Sign-up and login pages
  - The user should be able to register in the app with full name, email, and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- [ ]  Create a Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon, and the total amount of all the transactions that belong to that 
    category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - A button "add a new category" at the bottom brings the user to the page to create a new category.

- [ ]  Create a Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user can see the total amount for the category (the sum of all of the amounts of the transactions in that category).
  - A button "add a new transaction" at the bottom brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.
 

- [ ] Testing requirements
  - Create unit and integration tests for the most important components of the application.

- [ ] The link to the presentation:
  -  <a href="https://youtu.be/xP1vhA5a2bY"> Presentation Link </a>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,178 @@
...
+If you like this project please give it a star
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+A list of resources that i find helpful and would like to give credit to :
```
Comment (KDlamini) [inline README.md:159]: Kindly make sure that the README contains an "Acknowledgements" section or similar that gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).

File: app/views/groups/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+      <div class="std-color">
+        <p class="nav-menu-row"><strong class="nav-menu-row">Total: $<%= total_user_amount %></strong></p>
+        <% @groups.each do |group| %>
+          <p class="nav-menu-row"><%= group.name%>: <%= (100 * group.total_amount / total_user_amount).round(1) %>%</p>
+        <% end %>
+        <div><%= link_to "Edit this user", edit_user_path(current_user) %></div>
+        <%= button_to "SIGN OUT", destroy_user_session_path, method: :delete, class: "btn-signout" %>
+      </div>
+    </div>
```
Comment (KDlamini) [inline app/views/groups/index.html.erb]: Great job adding the delete and edit functionality on your category cards. You went above beyond 👏🏾🔥 

image

However, the card's look should follow the design specifications for the homepage. 

image

This means:

- A bar at the top of the page must display `total spending`.
- The three columns as illustrated above should have the `image icon` on the left, the `category name` and `date` in the middle, and the `price` on the right as per design specifications.

I would suggest that you rather have the `delete` and `edit` on the categories `show` page and remove them from the index page.

File: app/views/payments/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<nav class="group-header">
+  <%= link_to "BACK", group_url(@group_id), class: "link-header" %>
+  <p class="header-title">New transaction</p>
+  <p></p>
+</nav>
+
+<%= render "form", payment: @payment %>
```
Comment (KDlamini) [inline app/views/payments/new.html.erb:7]: When adding a new transaction for a category. There should be an option for choosing a category, ideally, a dropdown to select from that has a list of all the categories that were created.

image

Although, the functionality of your app automatically allocates the new transaction to the current category. It is still mandatory to provide an option for users to pick a different category.

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @odaialazzeh 🙋‍♀️,  

Great work so far! 💯 
You have done a great job creating a working Budget app using Rails 🎊 
Fantastic job on your application, you went above and beyond adding functionality to modify the user profile and categories 🎆 🍾 
There are some issues that you still need to work on improving your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞
 
✔️  Good PR title and description
✔️  Well done with adding user profile customization and modifying categories 🔥🔥🔥
✔️  Models, controllers, migrations, views, and routes set up correctly
✔️  Relationships and associations are well implemented
✔️  Project is deployed successfully and accessible online

## Video Presentation 🎥 

✔️ There is a recorded presentation video presenting the features of the project with a maximum length of 5 minutes.
✔️ The video contains a clear and objective description of the project in professional English. The topic and purpose of the project are made clear.
✔️ The video contains a demo of the project, highlighting all the project features.
✖️ The video contains a highlight about a piece of the code the student finds interesting or is proud of making.

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @odaialazzeh ,
# STATUS: APPROVED! 🥇 💯 
done

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**